test/test_distributions.py

```diff
@@ -3971,6 +3971,29 @@ def test_transformed_distribution_shapes(self):
             except NotImplementedError:
                 continue
 
+    def test_jit(self):
+        for transform in self.unique_transforms:
+            print('testing {}'.format(type(transform).__name__))
+            x = torch.tensor(self._generate_data(transform), requires_grad=True)
+            y = transform(x).clone()  # bypass the cache
+
+            def fwd(x):
+                return transform(x)
+
+            def inv(y):
+                return transform.inv(y)
+
+            def fwd_jacobian(x):
+                y = transform(x)
+                return transform.log_abs_det_jacobian(x, y)
+
+            try:
+                traced_fwd = torch.jit.trace(fwd, (x,))
+                traced_inv = torch.jit.trace(inv, (y,))
+                traced_fwd_jacobian = torch.jit.trace(fwd_jacobian, (x,))
```

Comment:
As I mentioned previously there are hardly any cases when `trace` would actually fail, but lack of errors doesn't imply that the trace is correct. There can be meaningful warnings, and it might even end up having a constant output. We should check that the traced functions can be reused with different values too, and return the same answers.

Reply:
Great idea! I've added stronger tests for all transforms.

Reply:
NB the same problem applies to all other non-sampling methods (`mean`, ...)